### ISSUE TITLE
mx:Alert Prevents Module From Loading in IE

### ISSUE THREAD

**Issue by yishayw** [apache/royale-asjs#826]
The main app:

```
<?xml version="1.0" encoding="utf-8"?>
<mx:Application xmlns:mx="library://ns.apache.org/royale/mx" xmlns:fx="http://ns.adobe.com/mxml/2009"
		   applicationComplete="init()">
	<fx:Script>
		<![CDATA[
			private function init():void
			{
				ml.url = "Test1.swf";
			}
		]]>
	</fx:Script>
	<mx:VBox width="600" height="600">
		<mx:ModuleLoader id="ml"/>
	</mx:VBox>
</mx:Application>

```

Test1.mxml :
```
<?xml version="1.0" encoding="utf-8"?>
<mx:Module xmlns:mx="library://ns.apache.org/royale/mx" 
	xmlns:fx="http://ns.adobe.com/mxml/2009" width="500" height="500">
	
<fx:Script>	<![CDATA[			
	import mx.controls.Alert; // delete this line to see module in IE
]]></fx:Script>
<mx:Label text="hello from module"/>
</mx:Module>
```

Removing the import means module is loaded, leaving it means it hangs.

I'm running this in debug mode.

**Comment by yishayw** [comment]
It's a mystery to me why dialogPolyfill.as is even referenced here. The only places I see dialogPolyfill referenced are in Jewel Alert and in mdl Dialog. Where is mx Alert getting this?

**Comment by aharui** [comment]
Good question.  Sometimes there are clues in the _deps.js and other .js files.  Search for polyfill and see how it gets referenced.

**Comment by aharui** [comment]
It was bugging me so I looked.  The compiler is not being smart about grabbing inject_html from externs.  It does not check whether an extern is actually used or not.

Still, though, we want to fix the pattern in case someone does use inject_html, so my suggesting to eliminate inject_html and replace with inject_js still seems like a good idea.

**Comment by yishayw** [comment]
Fixed with 14322f1
